### DISCUSSION TITLE
Use Zenodo + Citoid to enhance structured cites from a URL

### DISCUSSION THREAD

**Discussion by metasj** [knowledgefutures/pubpub#1316] [Ideas]
** Request **
_[Citoid](https://www.mediawiki.org/wiki/Citoid)_ is a node.js service that uses Zenodo and a handful of core citation templates (on Wikipedia) to autogenerate citations for a wide range of URLs, including news articles, books, articles, a wide range of pages that hold works with a DOI, and generic web pages.

Implementing something like this would let a much wider range of URLs autogenerate structured citations.  This is the single most popular editing feature that Wikimedia has ever developed.

**What would this feature help you accomplish, and why is that important to you?**
* As an editor, this would let me cite more quickly, meaningfully, beautifully.
* As an editor, I currently regularly get "Error" as the structured data output when entering a URL.  This would always produce some output, even if it's just a "web" template (title, url, date retrieved).

* As a reader, this would give me more meaningful and useful citations.

* As a reviewer, this would make it easier for me to parse what is being cited, as it normalized the format of citations across almost all use cases

### Optional
**Are there any other products or services that do this well?**
Zenodo itself handles DOI-style bibliographic data extraction.

Wikipedia's templates for a hundred different source-types are an improved distillation of APA-style citation style, optimized for the web (where you want certain parts of a citation to be hyperlinked to the source, want retrieved-on dates, and sometimes want links to wayback-machine archival links).

Citoid combines the above pretty cleanly, and includes sublibraries that may be useful even if citoid itself (a running service) is overkill.

**Comment by gabestein** [comment]
Given the work involved to setup a standalone service, I think we're unlikely to do this in the near future. But I'm going to move this to a discussion, because I do think at some point having something like this would be really useful, and folks may have other ideas for how to accomplish some smarter citations from URLs.
